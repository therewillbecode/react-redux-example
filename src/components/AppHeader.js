import React, { Component } from "react";
import { withRouter } from "react-router";
import { Sidebar, Menu, Icon } from "semantic-ui-react";

export class AppHeader extends Component {
  render() {
    const {
      isAuthenticated,
      login,
      logout,
      history,
      downloadInventoryCSV
    } = this.props;

    return (
      <div style={{ height: "5vh" }}>
        <Sidebar.Pushable
          as={Menu}
          direction="top"
          width="thin"
          visible
          borderless
          inverted
          style={{ background: "#085276" }}
        >
          <Menu.Item size="huge">
            <Icon name="barcode" bordered inverted size="big" />
          </Menu.Item>
          <Menu.Item link onClick={() => history.push("/")} name="title">
            <h2>Supply Chain Project</h2>
          </Menu.Item>
          {isAuthenticated ? (
            <Menu.Item
              link
              text
              onClick={() => history.push("/assets")}
              name="assets"
            >
              <h3>Assets</h3>
            </Menu.Item>
          ) : null}
          {!isAuthenticated ? (
            <Menu.Item link position="right" onClick={login} name="Login">
              <Icon name="sign in" />
              Login
            </Menu.Item>
          ) : null}
          {isAuthenticated ? (
            <Menu.Item
              link
              icon
              position="right"
              onClick={() => downloadInventoryCSV()}
              size="big"
              name="Download"
            >
              <Icon name="download" size="large" />
            </Menu.Item>
          ) : null}
          {isAuthenticated ? (
            <Menu.Item link size="huge" onClick={logout} name="Logout">
              <Icon name="sign out" size="large" />
              Logout
            </Menu.Item>
          ) : null}
        </Sidebar.Pushable>
      </div>
    );
  }
}

export default withRouter(AppHeader);
