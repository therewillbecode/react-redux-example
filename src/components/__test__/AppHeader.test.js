/* eslint-disable */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";

import { AppHeader } from "../AppHeader";

describe("AppHeader", () => {
  it("renders correctly when user is authenticated", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<AppHeader isAuthenticated={true} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when user is not authenticated", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<AppHeader isAuthenticated={false} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
