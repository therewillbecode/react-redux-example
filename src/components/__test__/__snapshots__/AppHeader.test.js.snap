// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders correctly 1`] = `
<div
  style={
    Object {
      "height": "5vh",
    }
  }
>
  <div
    className="ui borderless inverted pushable menu"
    direction="top"
    style={
      Object {
        "background": "#085276",
      }
    }
    visible={true}
    width="thin"
  >
    <div
      className="item"
      onClick={[Function]}
      size="huge"
    >
      <i
        aria-hidden="true"
        className="barcode big bordered inverted icon"
      />
    </div>
    <a
      className="link item"
      onClick={[Function]}
    >
      <h2>
        Supply Chain Project
      </h2>
    </a>
    <div
      className="right link item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="sign in icon"
      />
      Login
    </div>
  </div>
</div>
`;
